src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/Specifiable.cs

```diff
@@ -0,0 +1,45 @@
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common
+{
+    /// <summary>
+    /// Represents a value which may or may not be specified.
+    /// </summary>
+    /// <remarks>
+    /// This is independent of whether the value is null or not;
+    /// a value may be specified as null.
+    /// </remarks>
+    /// <typeparam name="T">The type of the value to be specified.</typeparam>
+    public struct Specifiable<T>
+    {
+        /// <summary>
+        /// Whether the value is specified.
+        /// </summary>
+        public bool IsSpecified;
```

Comment:
A public mutable field is almost certainly not what you intended here. 😀 

Suggest `public bool IsSpecified { get; }` instead.